test/integration/LimitOrderFeeCollector.t.sol

```diff
@@ -0,0 +1,413 @@
+
+        feeCollector.swap(settler, payable(address(this)), ETH, USDC, 1 wei, actions, bytes32(0));
+
+        assertGt(USDC.balanceOf(address(this)), 0);
+        assertEq(address(feeCollector).balance, 0);
+    }
+
+    function testMultiSwap() public {
+        deal(address(WETH), address(feeCollector), takingAmount);
+        deal(address(USDT), address(feeCollector), makingAmount);
+
+        bytes[] memory actionsOriginal = new bytes[](1);
+        actionsOriginal[0] = abi.encodeCall(
+            ISettlerActions.UNISWAPV3_VIP,
+            (
+                address(settler),
+                abi.encodePacked(WETH, uint8(0), uint24(500), USDC),
+                ISignatureTransfer.PermitTransferFrom({
+                    permitted: ISignatureTransfer.TokenPermissions({token: address(WETH), amount: type(uint256).max}),
+                    nonce: 0,
+                    deadline: block.timestamp + 5 minutes
+                }),
+                "",
+                0
+            )
+        );
+
+        bytes memory actions0 = abi.encode(actionsOriginal);
+        assembly ("memory-safe") {
+            let len := mload(actions0)
+            actions0 := add(0x20, actions0)
+            mstore(actions0, sub(len, 0x20))
+        }
+
+        actionsOriginal = new bytes[](2);
+        actionsOriginal[0] = abi.encodeCall(
+            ISettlerActions.UNISWAPV3_VIP,
+            (
+                address(settler),
+                abi.encodePacked(USDT, uint8(0), uint24(500), WETH),
+                ISignatureTransfer.PermitTransferFrom({
+                    permitted: ISignatureTransfer.TokenPermissions({token: address(USDT), amount: type(uint256).max}),
+                    nonce: 0,
+                    deadline: block.timestamp + 5 minutes
+                }),
+                "",
+                0
+            )
+        );
+        actionsOriginal[1] = abi.encodeCall(
+            ISettlerActions.BASIC,
+            (address(WETH), 10_000, address(WETH), 0x04, abi.encodeWithSignature("withdraw(uint256)", 0 wei))
+        );
+
+        bytes memory actions1 = abi.encode(actionsOriginal);
+        assembly ("memory-safe") {
+            let len := mload(actions1)
+            actions1 := add(0x20, actions1)
+            mstore(actions1, sub(len, 0x20))
+        }
+
+        Swap[] memory swaps = new Swap[](2);
+        swaps[0].recipient = payable(address(this));
+        swaps[0].sellToken = WETH;
+        swaps[0].buyToken = USDC;
+        swaps[0].minBuyAmount = 1 wei;
+        swaps[0].actions = actions0;
+        swaps[0].zid = bytes32(0);
+        swaps[1].recipient = payable(address(this));
+        swaps[1].sellToken = USDT;
+        swaps[1].buyToken = ETH;
+        swaps[1].minBuyAmount = 1 wei;
+        swaps[1].actions = actions1;
+        swaps[1].zid = bytes32(0);
+
+        uint256 beforeBalance = address(this).balance;
+
+        vm.expectEmit(false, true, true, false, address(USDC));
```

Comment:
We don't need to explicitly check for the second swap as its covered in the balance check, correct?

Reply:
There's no event for transferring ETH from Settler to the recipient (the test contract). I guess I could check for the event that transfers WETH from the pair to Settler or for the WETH `Withdraw` event. But yeah, it's covered by the check that the ETH balance of the test contract increases over the call to `multiSwap`. Plus I checked the trace by running the test suite with `-vvvv`